### PR TITLE
build: use custom command to create single versiondef

### PR DESCRIPTION
**Pull request by gpanders**
When using a multi config generator, CMake generates an output file for
each configuration when using file(GENERATE). When the contents of the
file for each configuration are different, CMake fails. Instead, create
separate files for each configuration and add a build time step to copy
the configuration specific file to the generic path "auto/versiondef.h"
which is included at build time.

### CODE REVIEW

**Review by jamessan — APPROVED** [review]

**Comment by dundargoc** [comment]
Sorry, I don't think this is correct. The behavior changes based on architecture and other variables. This currently fails on my build at home.

**Comment by gpanders** [comment]
@dundargoc I tried another approach using `add_custom_command`. We generate a variant of `versiondef-$CONFIG.h` for each config type and at build time copy the configuration being used to `auto/versiondef.h` (no suffix), which is picked up by the `#include` in `version.c`.

File: src/nvim/CMakeLists.txt [hunk 1]
```diff
@@ -466,6 +466,13 @@ add_custom_target(update_version_stamp
     -P ${PROJECT_SOURCE_DIR}/cmake/GenerateVersion.cmake
   BYPRODUCTS ${NVIM_VERSION_GIT_H})
 
+set(NVIM_VERSION_DEF_H ${PROJECT_BINARY_DIR}/cmake.config/auto/versiondef.h)
+add_custom_command(
+  OUTPUT "${NVIM_VERSION_DEF_H}"
+  COMMAND "${CMAKE_COMMAND}"
+  ARGS -E copy "${PROJECT_BINARY_DIR}/cmake.config/auto/versiondef-$<CONFIG>.h" "${NVIM_VERSION_DEF_H}"
```
Comment (dundargoc) [inline src/nvim/CMakeLists.txt:473]: Delete `ARGS`, it's legacy.

Comment (gpanders) [reply src/nvim/CMakeLists.txt]: Removed 548c99e0d075662ec06abd2e6c1f65ddfde961d8

**Pull request merged**